### DISCUSSION TITLE
What's the use case for intersect over merge?

### DISCUSSION THREAD

**Discussion by npearson72** [colinhacks/zod#2703] [Q&A]
I was a little confused by this part of the documentation:

> Though in many cases, it is recommended to use A.merge(B) to merge two objects. The .merge method returns a new ZodObject instance, whereas A.and(B) returns a less useful ZodIntersection instance that lacks common object methods like pick and omit.

When would I ever prefer the "less useful" `ZodIntersection` instance over the `ZodObject`?

**Comment by mnahkies** [comment]
One example that came up for me today is I wanted to combine an object with a record, which required me to use `intersection` instead of `merge`, eg:
```
const s = z.intersection(z.object({foo: z.string()}), z.record(z.number()))
type t = z.infer<typeof s>

// type t = {
//    foo: string;
// } & Record<string, number>
```
